### PR TITLE
Improved handling of span status

### PR DESCRIPTION
**Pull request by antonpirker**
Having the Sentry span statuses in a class and introducing `get_span_status_from_http_code()`. This will be used in our new OpenTelementry instrumentation. 

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: sentry_sdk/consts.py [hunk 1]
```diff
@@ -386,6 +386,32 @@ class SPANDATA:
     """
 
 
+class SPANSTATUS:
```
Comment (szokeasaurusrex) [inline sentry_sdk/consts.py:389]: How about we make this an [enum](https://docs.python.org/3/library/enum.html#enum.Enum)?

File: sentry_sdk/tracing.py [hunk 2]
```diff
@@ -151,6 +151,45 @@ class TransactionKwargs(SpanKwargs, total=False):
...
+
+    elif 500 <= http_status_code < 600:
+        if http_status_code == 504:
+            return SPANSTATUS.DEADLINE_EXCEEDED
+        elif http_status_code == 501:
+            return SPANSTATUS.UNIMPLEMENTED
+        elif http_status_code == 503:
+            return SPANSTATUS.UNAVAILABLE
+        else:
+            return SPANSTATUS.INTERNAL_ERROR
```
Comment (szokeasaurusrex) [inline sentry_sdk/tracing.py:188]: Would be nice to sort the if statements in increasing order by status code. Although, perhaps this would better belong in a separate PR

**Review by szokeasaurusrex — APPROVED** [review]
Looks good, please consider suggestions before merging

**Comment by antonpirker** [comment]
Thanks for the quick review @szokeasaurusrex 
I will keep it as is for now and just merge, do not want to change to much right now, we can always improve later.
